### ISSUE TITLE
After upgrading SAM CLI - Docker StdOut appears to be buffered

### ISSUE THREAD

**Issue by singulli1** [awslabs/aws-sam-cli#835]
### Description

Windows 10 64-bit
SAM CLI, version 0.7.0
Docker version 18.09.0, build 4d60db4
Runtime: node.js 6.10 

Before the upgrade, everything worked great.
After upgrade, it appears stdout from my application running in my docker image is buffering until the application completes, at which time stdout appears to be flushed and I can see it.

### Steps to reproduce

1. open windows cmd prompt.
2. navigate to node.js project directory
3. run sam local invoke cmd

example cmd:
sam local invoke  -e ./src/scheduledEvent.json  "ClaimMDIntegration" --template templateDev.yaml

In my node.js application, there are several console.log("Messages") commands.  I'd expect to see messages written to my command prompt window as they occur.

### Observed result
Instead, messages are written to the command prompt only after the program has completed execution.

### Expected result

Message should show up in command prompt as they occur, during program execution.  This was the behavior before I upgraded.

`Add --debug flag to command you are running`

C:\Users\scott\projectsSAM\psykdeskClaimsIntegration>sam local invoke --debug  -e ./src/scheduledEvent.json  "ClaimMDIntegration" --template templateDev.yaml
2018-12-06 14:06:43 Using SAM Template at C:\Users\scott\projectsSAM\psykdeskClaimsIntegration\templateDev.yaml
2018-12-06 14:06:43 local invoke command is called
2018-12-06 14:06:43 No Parameters detected in the template
2018-12-06 14:06:43 1 resources found in the template
2018-12-06 14:06:43 Found Serverless function with name='ClaimMDIntegration' and CodeUri='./build'
2018-12-06 14:06:43 Trying paths: ['C:\\Users\\scott\\.docker\\config.json', 'C:\\Users\\scott\\.dockercfg']
2018-12-06 14:06:43 Found file at path: C:\Users\scott\.docker\config.json
2018-12-06 14:06:43 Couldn't find auth-related section ; attempting to interpretas auth-only file
2018-12-06 14:06:43 Config entry for key stackOrchestrator is not auth config
2018-12-06 14:06:43 http://localhost:None "GET /v1.35/_ping HTTP/1.1" 200 2
2018-12-06 14:06:43 Trying paths: ['C:\\Users\\scott\\.docker\\config.json', 'C:\\Users\\scott\\.dockercfg']
2018-12-06 14:06:43 Found file at path: C:\Users\scott\.docker\config.json
2018-12-06 14:06:43 Couldn't find auth-related section ; attempting to interpretas auth-only file
2018-12-06 14:06:43 Config entry for key stackOrchestrator is not auth config
2018-12-06 14:06:43 Found one Lambda function with name 'ClaimMDIntegration'
2018-12-06 14:06:43 Invoking claimMDIntegration.handler (nodejs6.10)
2018-12-06 14:06:43 Environment variables overrides data is standard format
2018-12-06 14:06:43 Loading AWS credentials from session with profile 'None'
2018-12-06 14:06:43 Changing event name from creating-client-class.iot-data to creating-client-class.iot-data-plane
2018-12-06 14:06:43 Changing event name from before-call.apigateway to before-call.api-gateway
2018-12-06 14:06:43 Changing event name from request-created.machinelearning.Predict to request-created.machine-learning.Predict
2018-12-06 14:06:43 Changing event name from before-parameter-build.autoscaling.CreateLaunchConfiguration to before-parameter-build.auto-scaling.CreateLaunchConfiguration
2018-12-06 14:06:43 Changing event name from before-parameter-build.route53 to before-parameter-build.route-53
2018-12-06 14:06:43 Changing event name from request-created.cloudsearchdomain.Search to request-created.cloudsearch-domain.Search
2018-12-06 14:06:43 Changing event name from docs.*.autoscaling.CreateLaunchConfiguration.complete-section to docs.*.auto-scaling.CreateLaunchConfiguration.complete-section
2018-12-06 14:06:43 Changing event name from before-parameter-build.logs.CreateExportTask to before-parameter-build.cloudwatch-logs.CreateExportTask
2018-12-06 14:06:43 Changing event name from docs.*.logs.CreateExportTask.complete-section to docs.*.cloudwatch-logs.CreateExportTask.complete-section
2018-12-06 14:06:43 Changing event name from before-parameter-build.cloudsearchdomain.Search to before-parameter-build.cloudsearch-domain.Search
2018-12-06 14:06:43 Changing event name from docs.*.cloudsearchdomain.Search.complete-section to docs.*.cloudsearch-domain.Search.complete-section
2018-12-06 14:06:43 Looking for credentials via: env
2018-12-06 14:06:43 Looking for credentials via: assume-role
2018-12-06 14:06:43 Looking for credentials via: shared-credentials-file
2018-12-06 14:06:43 Found credentials in shared credentials file: ~/.aws/credentials
2018-12-06 14:06:43 Resolving code path. Cwd=C:\Users\scott\projectsSAM\psykdeskClaimsIntegration, CodeUri=./build
2018-12-06 14:06:43 Resolved absolute path to code is C:\Users\scott\projectsSAM\psykdeskClaimsIntegration\build
2018-12-06 14:06:43 Code C:\Users\scott\projectsSAM\psykdeskClaimsIntegration\build is not a zip/jar file
2018-12-06 14:06:43 Trying paths: ['C:\\Users\\scott\\.docker\\config.json', 'C:\\Users\\scott\\.dockercfg']
2018-12-06 14:06:43 Found file at path: C:\Users\scott\.docker\config.json
2018-12-06 14:06:43 Couldn't find auth-related section ; attempting to interpretas auth-only file
2018-12-06 14:06:43 Config entry for key stackOrchestrator is not auth config
2018-12-06 14:06:43 http://localhost:None "GET /v1.35/images/lambci/lambda:nodejs6.10/json HTTP/1.1" 200 None
2018-12-06 14:06:43 Looking for auth config
2018-12-06 14:06:43 Looking for auth entry for 'docker.io'
2018-12-06 14:06:43 No entry found
2018-12-06 14:06:43 No auth config found
2018-12-06 14:06:44 http://localhost:None "POST /v1.35/images/create?tag=nodejs6.10&fromImage=lambci%2Flambda HTTP/1.1" 200 None

Fetching lambci/lambda:nodejs6.10 Docker container image......
2018-12-06 14:06:44 Mounting C:\Users\scott\projectsSAM\psykdeskClaimsIntegration\build as /var/task:ro inside runtime container
2018-12-06 14:06:44 http://localhost:None "POST /v1.35/containers/create HTTP/1.1" 201 90
2018-12-06 14:06:44 http://localhost:None "GET /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/json HTTP/1.1" 200 None
2018-12-06 14:06:44 http://localhost:None "GET /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/json HTTP/1.1" 200 None
2018-12-06 14:06:45 http://localhost:None "POST /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/start HTTP/1.1" 204 0
2018-12-06 14:06:45 Starting a timer for 300 seconds for function 'ClaimMDIntegration'
2018-12-06 14:06:45 http://localhost:None "GET /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/json HTTP/1.1" 200 None
2018-12-06 14:06:45 http://localhost:None "POST /containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/attach?stdout=1&stderr=1&logs=1&stream=1&stdin=0 HTTP/1.1" 101 0
[32mSTART RequestId: 1c59cf45-add4-120f-a076-27d09f5ca084 Version: $LATEST[0m
[34m2018-12-06T19:06:48.396Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0m{"account":"123456789012","region":"us-east-1","detail":{},"detail-type":"Scheduled Event","source":"aws.events","time":"1970-01-01T00:00:00Z","id":"cdc73f9d-aea9-11e3-9d5a-835b769c0d9c","TenantID":"1","resources":["arn:aws:events:us-east-1:123456789012:rule/my-schedule"]}
[34m2018-12-06T19:06:49.633Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mIs Production: false
[34m2018-12-06T19:06:49.633Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mdataGateway host: kaqun47c34.execute-api.us-east-2.amazonaws.com
[34m2018-12-06T19:06:49.633Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mdataGateway path: /Dev/dataGateway
[34m2018-12-06T19:06:49.634Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetTenant
[34m2018-12-06T19:06:50.064Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetItems - Users
[34m2018-12-06T19:06:50.326Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetItems - Insurances
[34m2018-12-06T19:06:50.602Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mcreateItem - BatchProcesses
[34m2018-12-06T19:06:50.825Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mprocessSubmittedClaims
[34m2018-12-06T19:06:50.826Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetItems - Claims
[34m2018-12-06T19:06:51.120Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mNo submitted claims
[34m2018-12-06T19:06:51.120Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mupdateItems - BatchProcesses
[34m2018-12-06T19:06:51.392Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mcreateItem - BatchProcesses
[34m2018-12-06T19:06:51.662Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mget reponses from claim.md
[34m2018-12-06T19:06:51.663Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetTenantCounters
[34m2018-12-06T19:06:51.976Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mlast Claim ResponseID: 265363497
[34m2018-12-06T19:06:51.977Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetNextTenantCounterNumber - RemoteFileCounter
[34m2018-12-06T19:06:53.234Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mresponse count: 0
[34m2018-12-06T19:06:53.234Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mupdateItems - BatchProcesses
[34m2018-12-06T19:06:53.460Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mcreateItem - BatchProcesses
[34m2018-12-06T19:06:53.721Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mget Eras from claim.md
[34m2018-12-06T19:06:53.721Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetTenantCounters
[34m2018-12-06T19:06:53.897Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mlast ERA Id: 7539961
[34m2018-12-06T19:06:53.897Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mgetNextTenantCounterNumber - RemoteFileCounter
[34m2018-12-06T19:06:55.006Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mERA count: 0
[34m2018-12-06T19:06:55.006Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0mupdateItems - BatchProcesses
[34m2018-12-06T19:06:55.290Z   1c59cf45-add4-120f-a076-27d09f5ca084    [0msuccess
[32mEND RequestId: 1c59cf45-add4-120f-a076-27d09f5ca084[0m
[32mREPORT RequestId: 1c59cf45-add4-120f-a076-27d09f5ca084     Duration: 9671.58 ms    Billed Duration: 9700 ms        Memory Size: 128 MB       Max Memory Used: 64 MB  [0m
2018-12-06 14:06:55 http://localhost:None "GET /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42/json HTTP/1.1" 200 None
2018-12-06 14:06:56 http://localhost:None "DELETE /v1.35/containers/a26fa40fbd80c31230543d9b8a2a7f9a68e90932fdf5566666b9581a43092f42?v=False&link=False&force=True HTTP/1.1" 204 0

{"statusCode":"Success","result":"success"}

**Comment by ndobryanskyy** [comment]
Hello @singulli1 I am working on the fix for that

**Comment by singulli1** [comment]
Cool thanks much! :)  Is there another ticket tracking this?
Thanks,

**Comment by ndobryanskyy** [comment]
#729 was trying to address similar issue, and also I've came across this problem and documented in #759
